File: app/src/main/java/com/quoders/apps/madridbus/ui/stopInfo/StopInfoPresenterModule.java
package com.quoders.apps.madridbus.ui.stopInfo;

import dagger.Module;
import dagger.Provides;

@Module
public class StopInfoPresenterModule {

    private final StopInfoContract.View mView;

    public StopInfoPresenterModule(StopInfoContract.View view) {
        mView = view;
    }

    @Provides
    StopInfoContract.View provideLinesContractView() {
        return mView;
    }
}

File: app/src/main/java/com/quoders/apps/madridbus/ui/stopInfo/StopInfoPresenterModule.kt
package com.quoders.apps.madridbus.ui.stopInfo

import dagger.Module
import dagger.Provides

@Module
class StopInfoPresenterModule(private val mView: StopInfoContract.View) {

    @Provides
    internal fun provideStopInfoContractView(): StopInfoContract.View {
        return mView
    }
}
